### DISCUSSION TITLE
TRV10 - v2.0.1 - patch version

### DISCUSSION THREAD

**Discussion by Rishabh-ondc** [ONDC-Official/mobility-specification#184] [General]
Features addressed as part of the v2.0.1 patch release are mentioned below. 

Fulfilment.type ENUM - 'SELF_PICKUP'
- This is to handle Queueing use cases at designated locations like airports, railway stations etc.
- The journey difference from protocol point of view is that for such fulfilment as part of on_confirm/ on_update once the state is 'RIDE_CONFIRMED', only OTP will be shared but no agent/ vehicle details; it will be shared along with 'RIDE_ASSIGNED'
- There may be instructions in the 'START' stop type to reach the respective pickup point which would be required to be displayed on the buyer app<br>

Fulfilment.vehicle.category ENUMs - 'CAB', 'TWO_WHEELER', 'AUTO_RICKSHAW'<br>

Fulfilment.vehicle.variant ENUMs
- AUTO_RICKSHAW: AUTO_RICKSHAW
- CAB: SEDAN, SUV, HATCHBACK 
- TWO_WHEELER: TWO_WHEELER

Describing the energy type of the vehicle, vehicle.energy_type ENUMs - 
- BIOFUELS
- CNG
- DIESEL
- ELECTRIC
- HYDROGEN
- LPG
- PETROL

Cancellation reason code ENUMs defined:
- Technical cancellation (BAP/BPP) - 000 : Flow added
- Driver not moving (BAP) - 001
- Unable to contact driver (BAP) - 002
- Driver asked me to cancel (BAP) - 003
- My pickup location is incorrect (BAP) - 004
- Booked by mistake (BAP) - 005
- No drivers available (BPP) - 011
- Rider not present at location (BPP) - 012
- Accepted by mistake (BPP) - 013
- Unable to contact rider (BPP) - 014
- Stopped by traffic officials (BPP) - 015
- Vehicle issue (punctured tyre, engine issues) (BPP) - 016

Fulfilment.state.descriptor.code ENUM -  'RIDE_CONFIRMED'
- RIDE_CONFIRMED - Whenever a ride is confirmed seller app will share this ENUM this is before RIDE_ASSIGNED
- Order of fulfilment status updates to be followed:
  - For 'DELIVERY' fulfilment Type - Happy Flow
    1. RIDE_CONFIRMED
    2. RIDE_ASSIGNED
    3. RIDE_ENROUTE_PICKUP
    4. RIDE_ARRIVED_PICKUP
    5. RIDE_STARTED
    6. RIDE_ENDED
  - For 'DELIVERY' fulfilment Type - Cancel Flow 
    1. RIDE_CONFIRMED
    2. RIDE_ASSIGNED
    3. RIDE_ENROUTE_PICKUP (Optional)
    4. RIDE_ARRIVED_PICKUP (Optional)
    5. RIDE_CANCELLED
  - For 'SELF_PICKUP' fulfilment Type - Happy Flow
    1. RIDE_CONFIRMED
    2. RIDE_ASSIGNED
    3. RIDE_STARTED
    4. RIDE_ENDED

**Comment by mherle** [comment]
> Fulfilment.vehicle.category ENUMs - 'CAB','TWO_WHEELER' 

It should also have AUTO_RICKSHAW, right ?

**Reply by Rishabh-ondc** [reply]
> > Fulfilment.vehicle.category ENUMs - 'CAB','TWO_WHEELER'
> 
> It should also have AUTO_RICKSHAW, right ?

Yes, updated

**Comment by mherle** [comment]
@Rishabh-ondc what's the real world meaning for the RIDE_ENROUTE_PICKUP ? Isn't the expectation that the rider is starting when assigned ?

In case of a booking where the assigned rider is dropping a previous customer, should this come at the time of acceptance itself or after the previous rider is completed ?

**Reply by Rishabh-ondc** [reply]
> @Rishabh-ondc what's the real world meaning for the RIDE_ENROUTE_PICKUP ? Isn't the expectation that the rider is starting when assigned ?

ASSIGNED and ENRUOTE_PICKUP are considered as separate states. 

> In case of a booking where the assigned rider is dropping a previous customer, should this come at the time of acceptance itself or after the previous rider is completed ?

After the previous ride is completed ideally

**Comment by mherle** [comment]
@Rishabh-ondc 
Need two more cases of cancellation by rider:

- Caught by police/traffic authorities
- Vehicle issues (tyre puncture, not starting etc)
